pilot/helpers/agents/SpecWriter.py

```diff
@@ -0,0 +1,71 @@
+from helpers.AgentConvo import AgentConvo
+from helpers.Agent import Agent
+from utils.files import count_lines_of_code
+from utils.style import color_green_bold, color_yellow_bold
+from prompts.prompts import ask_user
+
+INITIAL_PROJECT_HOWTO_URL = "https://github.com/Pythagora-io/gpt-pilot/wiki/How-to-write-a-good-initial-project-description"
+
+class SpecWriter(Agent):
+    def __init__(self, project):
+        super().__init__('spec_writer', project)
+        self.save_dev_steps = True
+
+    def analyze_project(self, initial_prompt):
+        msg = (
+            "Your project description seems a bit short. "
+            "The better you can describe the project, the better GPT Pilot will understand what you'd like to build.\n\n"
+            f"Here are some tips on how to better describe the project: {INITIAL_PROJECT_HOWTO_URL}\n\n"
+        )
+        print(color_yellow_bold(msg))
+        print(color_green_bold("Let's start by refining your project idea:"))
+
+        convo = AgentConvo(self)
+        convo.construct_and_add_message_from_prompt('spec_writer/ask_questions.prompt', {})
+
+        user_response = initial_prompt
+        while True:
+            llm_response = convo.send_message('utils/python_string.prompt', {
+                "content": user_response,
+            })
+            if not llm_response:
+                continue
+
+            llm_response = llm_response.strip()
+            if len(llm_response) > 500:
+                user_response = ask_user(
```

Comment:
We are missing print button

<img width="1047" alt="Screenshot 2024-02-16 at 5 10 41 PM" src="https://github.com/Pythagora-io/gpt-pilot/assets/41999013/da245206-0f6c-4ca0-a604-671a90d2a306">

Reply:
![Screenshot from 2024-02-16 18-21-33](https://github.com/Pythagora-io/gpt-pilot/assets/3362/417c9fe9-81b5-4ee8-8a16-aef8976318ce)